src/ClientProgram.elm

```diff
@@ -79,15 +79,20 @@ update :
     -> ( Model, Cmd Msg )
 update ports msg model =
     case msg of
-        Unknown value ->
-            ( model, logWarning ("No handler for unknown message: " ++ toString value) )
-
         ClientMsg message ->
             handleAppMessage ports.toHost message model
 
         HostMsg message ->
             handleHostMessage ports.toClient message model
 
+        BadMessage error ->
+            ( model
+            , logWarning
+                ("Could not parse incoming message: "
+                    ++ Decode.errorToString error
```

Comment:
JSON decoding errors in Elm now produce structured data instead of just a string.  While I was dealing with that, I changed `Unknown` to `BadMessage` which I think more accurately represents what's going on in this case (we got pushed some data that we can't make sense of).